src/python/pants/base/mustache.py

```diff
@@ -34,7 +35,7 @@ def convert_val(x):
         return [convert_val(e) for e in x]
       else:
         return x
-    items = [(key, convert_val(val)) for (key, val) in args.items()]
+    items = [(key, convert_val(val)) for (key, val) in list(args.items())]
```

Comment:
I guess these are automated, but they are unfortunate. An Iterable is enough to build a list comprehension and so wrapping in a list is a waste of resources and extra un-needed code noise.

Reply:
Agreed. We can manually try to identify places like this if you think it's worth it, although will slow us down and may be error prone knowing when and when not to change back.

Reply:
You can definitely safely remove it in this context - ie: immediate consumption of the Iterable. That said, I loathe scaning for these in the conversion PRs - does not scale and we'll likely mess up. It would be awesome if the tool was better or configurable to avoid this case. Presumably it uses an AST and not just string searches. If it does use an AST, a super-nice contribution would be expanding about the dict.{items,keys,values} call to see if the context is the generator of a for loop, and if so - do not wrap.

Reply:
@jsirois I've been reviewing the changes already - not that much extra work for me to remove `list()` when it's used in for loops. Will make your lives easier with PRs also - less changes.

Maintainers stopped updating `futurize`, `2to3`, etc last year, so PR wouldn't go far there.